app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -314,15 +314,29 @@ private File createAvatarFileFromStudentImage(StudentImage studentImage, Student
 
     /**
      * Find similar students
-     * Case 1: Student was added during fallback but in the meantime the same person has an existing StudentImageCollectionEvent and a new Student entry
+     *
+     *
      */
     public synchronized void findAndMergeSimilarStudents(){
         Log.i(getClass().getName(), "findAndMergeSimilarStudents");
         PreProcessorFactory ppF = new PreProcessorFactory(context);
+        TensorFlow tensorFlow = getInitializedTensorFlow();
+        findSimilarStudentsUsingAvatarImages(ppF, tensorFlow);
+        findSimilarStudentsUsingMeanFeatureVector(ppF, tensorFlow);
+    }
+
+    /**
+     * Find similar students
+     * Case 1: Student was added during fallback but in the meantime the same person has an existing StudentImageCollectionEvent and a new Student entry
+     * ---> Use the avatar image as input for the recognition
+     * @param ppF
+     * @param tensorFlow
+     */
+    private synchronized void findSimilarStudentsUsingAvatarImages(PreProcessorFactory ppF, TensorFlow tensorFlow){
+        // Iterate through all Students
         List<Student> students = studentDao.loadAll();
```

Comment:
Add logging here so that we can see if the method was triggered

`Log.i(getClass().getName(), "findSimilarStudentsUsingAvatarImages");`